connect-inject/envoy_sidecar.go

```diff
@@ -21,6 +21,13 @@ func (h *Handler) envoySidecar(pod corev1.Pod, k8sNamespace string) (corev1.Cont
 		return corev1.Container{}, err
 	}
 
+	if pod.Spec.SecurityContext != nil {
+		// User container and Envoy container cannot have the same UID.
+		if pod.Spec.SecurityContext.RunAsUser != nil && *pod.Spec.SecurityContext.RunAsUser == envoyUserAndGroupID {
```

Comment:
it looks we're only checking the security context of the pod spec, but we probably should also check the security context of the user container.

Reply:
Ah right, I completely forgot about this, thanks for double checking. Will add this!